site/src/modules/resources/PortForwardButton.tsx

```diff
@@ -26,24 +30,45 @@ import {
   PopoverTrigger,
 } from "components/Popover/Popover";
 import KeyboardArrowDown from "@mui/icons-material/KeyboardArrowDown";
+import Stack from "@mui/material/Stack";
+import Select from "@mui/material/Select";
+import MenuItem from "@mui/material/MenuItem";
+import FormControl from "@mui/material/FormControl";
+import TextField from "@mui/material/TextField";
+import SensorsIcon from "@mui/icons-material/Sensors";
+import LockIcon from "@mui/icons-material/Lock";
+import LockOpenIcon from "@mui/icons-material/LockOpen";
+import CloseIcon from "@mui/icons-material/Close";
+import {
+  deleteWorkspacePortShare,
+  upsertWorkspacePortShare,
+  workspacePortShares,
+} from "api/queries/workspaceportsharing";
+import { useDashboard } from "modules/dashboard/useDashboard";
+import * as Yup from "yup";
+import { FormikContextType, useFormik } from "formik";
+import { getFormHelpers } from "utils/formUtils";
 
 export interface PortForwardButtonProps {
   host: string;
   username: string;
   workspaceName: string;
+  workspaceID: string;
   agent: WorkspaceAgent;
+  template: Template;
 
   /**
    * Only for use in Storybook
    */
   storybook?: {
-    portsQueryData?: WorkspaceAgentListeningPortsResponse;
+    listeningPortsQueryData?: WorkspaceAgentListeningPortsResponse;
+    sharedPortsQueryData?: WorkspaceAgentPortShares;
```

Comment:
We don't need these for storybook only since you can mock queries there:

```tsx
const SomeStory: Story = {
  parameters: {
    queries: [
      {
        key: listeningPortsQueryKey, // Query key
        data: [...], // Mock data
      },
    ],
  }
}
```

Reply:
Awesome!